experimental/ag-ui/server.py

```
@@ -0,0 +1,411 @@
+                logging.warning(f"Failed to parse result as JSON: {result_data}")
+                result_data = {}
+
+        # Handle different Prometheus response formats
+        prometheus_data = result_data
+        result_type = "unknown"
+        if "data" in result_data:
+            prometheus_data = json.loads(result_data["data"]).get("data")
+            result_type = prometheus_data.get("resultType", "unknown")
+
+        # Generate a meaningful title
+        title = f"Prometheus Query Results"
```

Ruff (0.13.3)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)